pkg/controller/buildrun/buildrun_controller.go

```diff
@@ -299,10 +298,14 @@ func (r *ReconcileBuildRun) retrieveTaskRun(build *buildv1alpha1.Build, buildRun
 	return nil, nil
 }
 
-func (r *ReconcileBuildRun) updateBuildRunErrorStatus(buildRun *buildv1alpha1.BuildRun, err error) *buildv1alpha1.BuildRun {
+func (r *ReconcileBuildRun) updateBuildRunErrorStatus(buildRun *buildv1alpha1.BuildRun, err error) error {
 	buildRun.Status.Succeeded = corev1.ConditionFalse
 	buildRun.Status.Reason = err.Error()
 	now := metav1.Now()
 	buildRun.Status.StartTime = &now
-	return buildRun
+	updateErr := r.client.Status().Update(context.TODO(), buildRun)
+	if updateErr != nil {
+		return updateErr
+	}
+	return err
```

Comment:
this is strange, while you get `err` and then you return the same `err`. This is why I suggest on my above comment to get rather an `string` as the func() parameter. The only error that you should be returning is the one from the client Status update call.

Reply:
Like the answer above, it should not return a string. And based on the discussion above, my initial logic is:
- get error
- update buildrun status to show error
- if update status wrong, report update status error, reconcile
- if update status successfully, return the error for reconcile

That is why the func named as `updateBuildRunErrorStatus`

Reply:
Maybe my comment was not clear enough, but I just wrote you the whole function definition with what I meant.